### PR TITLE
compose apps; allow apps to call apps + start the big SDK refactoring + many UI improvements

### PR DESCRIPTION
**Pull request by rvion**

### CODE REVIEW

File: library/sdk-examples/example-custom-store.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ app({
     }),
 
     run: async (sdk, ui) => {
-        const store = sdk.getStore_orCreateIfMissing('example-key-1337', () => ({ count: 0 }))
+        const store = sdk.store.get_orCreate({ key: 'example-key-1337', makeDefaultValue: () => ({ count: 0 }) })
```
Comment (ricklove) [inline library/sdk-examples/example-custom-store.ts:16]: With the shorter name, `getOrCreate` seems better

File: src/cards/LibraryFile.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { ManualPromise } from '../utils/misc/ManualPromise'
 import { getPngMetadataFromUint8Array } from '../utils/png/_getPngMetadata'
 import { Library } from './Library'
 
-import { dirname } from 'path'
+import { dirname } from 'pathe'
```
Comment (ricklove) [inline src/cards/LibraryFile.ts:15]: pathe? - oh that is a nice library, I'll have to start using that

File: src/models/CushyScriptL.ts [hunk 3]
```diff
@@ -109,12 +109,16 @@ export class CushyScriptL {
...
         // 1. setup DI registering mechanism
-        const registerAppFn = (a1: string, a2: App<any>): void => {
+        const registerAppFn = (a1: string, a2: App<any>): AppRef<any> => {
             const app: App<WidgetDict> = typeof a1 !== 'string' ? a1 : a2
             const name = app.metadata?.name ?? basename(this.relPath)
             console.info(`[💙] found action: "${name}"`, { path: this.relPath })
             APPS.push(app)
+            const appID = asCushyAppID(this.relPath + ':' + appIndex++) // 🔴 SUPER UNSAFE
+            // console.log(`[👙] >> appID==`, appID)
+            return { $DO_NOT_USE: 0 as any, id: appID }
```
Comment (ricklove) [inline src/models/CushyScriptL.ts:121]: Why not just use `$Output`? I think the `$` is sufficient to let people know to leave it alone

File: src/models/Step.ts [hunk 4]
```diff
@@ -18,11 +18,13 @@ import { MediaSplatL } from './MediaSplat'
 import { MediaTextL } from './MediaText'
 import { MediaVideoL } from './MediaVideo'
 import { RuntimeErrorL } from './RuntimeError'
+import { ManualPromise } from 'src/utils/misc/ManualPromise'
 
 export type FormPath = (string | number)[]
 /** a thin wrapper around an app execution */
 export interface StepL extends LiveInstance<StepT, StepL> {}
 export class StepL {
+    completionPromise = new ManualPromise<RuntimeExecutionResult>()
```
Comment (ricklove) [inline src/models/Step.ts:27]: Perhaps: `finished` to match `ComfyPromptL`

**Review by ricklove — COMMENTED** [review]
Nice work! This feels like a big improvement!

Added some naming suggestions

File: src/cards/LibraryFile.ts (hunk 2 above)
Comment (rvion) [reply src/cards/LibraryFile.ts:15]: just replaced all remaining path a few minutes ago; I'll try to add some lint rule that prevent using it; or maybe I'll just alias path to pathe directly instead :)

**Comment by rvion** [comment]
great suggestions and help on this, thanks @ricklove ! 🔥

**Pull request merged**